### PR TITLE
vim: Increment search right

### PR DESCRIPTION
**Pull request by hoodie**
Hi there, nice editor!
Here's my attempt at fixing #10865.

Thanks

Release Notes:

-vim: Fix ctrl+a when cursor is on a decimal point ([#10865](https://github.com/zed-industries/zed/issues/10865)).

### CODE REVIEW

**Comment by hoodie** [comment]
@cla-bot check'

File: crates/vim/src/normal/increment.rs [hunk 1]
```diff
@@ -117,13 +117,16 @@ fn find_number(
...
 
-    // go backwards to the start of any number the selection is within
-    for ch in snapshot.reversed_chars_at(offset) {
-        if ch.is_ascii_digit() || ch == '-' || ch == 'b' || ch == 'x' {
-            offset -= ch.len_utf8();
-            continue;
+    let ch0 = snapshot.chars_at(offset).next();
+    if ch0.as_ref().is_some_and(char::is_ascii_digit) || matches!(ch0, Some('-' | 'b' | 'x')) {
+        // go backwards to the start of any number the selection is within
+        for ch in snapshot.reversed_chars_at(offset) {
```
Comment (hoodie) [inline crates/vim/src/normal/increment.rs:123]: there's probably a nicer way to reverse iterate that avoids this extra if in line 121

**Comment by hoodie** [comment]
is the dangerJS warning supposed to remain?

**Comment by ConradIrwin** [comment]
@hoodie Thank you for this (and for adding a test), will merge when CI passes.

(also, if you're on a roll and excited to continue improving `ctrl-a`/`ctrl-x`, https://github.com/zed-industries/zed/issues/10400 has a request for `g ctrl-a` which is in vim, and for toggling a bool value, which seems nice to have).

**Pull request merged**